core/state/snapshot/disklayer.go

```diff
@@ -133,6 +181,50 @@ func (dl *diskLayer) AccountRLP(hash common.Hash) ([]byte, error) {
 func (dl *diskLayer) Storage(accountHash, storageHash common.Hash) ([]byte, error) {
 	dl.lock.RLock()
 	defer dl.lock.RUnlock()
+	start := time.Now()
+
+	routeid := cachemetrics.Goid()
+	hitInL3 := false
+	hitInDisk := false
+	var startGetInDisk time.Time
+	defer func() {
+		isSyncMainProcess := cachemetrics.IsSyncMainRoutineID(routeid)
+		isMinerMainProcess := cachemetrics.IsMinerMainRoutineID(routeid)
+		if isSyncMainProcess {
+			// layer 2 miss
+			syncL2StorageMissMeter.Mark(1)
+			if hitInL3 {
```

Comment:
move this `syncL2StorageMissMeter.Mark(1)` into `if hitInL3`

Reply:
if we reach this function, it can be concluded that  L2 layer already missed , so it does not matter whether layer3 hit or miss.